cmd/run_java.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"go.temporal.io/sdk-features/harness/go/cmd"
+	"go.temporal.io/server/common/log/tag"
+)
+
+// RunJavaExternal runs the Java run in an external process. This expects the
+// server to already be started.
+func (r *Runner) RunJavaExternal(ctx context.Context, run *cmd.Run) error {
+	// To do this, we're gonna create a temporary project, --include-build this
+	// one, then "gradle run" it
+
+	// Create base dir
+	_, currFile, _, _ := runtime.Caller(0)
+	rootDir := filepath.Dir(filepath.Dir(currFile))
+	tempDir, err := os.MkdirTemp(rootDir, "sdk-features-java-test-")
+	if err != nil {
+		return fmt.Errorf("failed creating temp dir: %w", err)
+	}
+	r.log.Info("Building temporary Java project", tag.NewStringTag("Path", tempDir))
+	// Remove when done if configured to do so
+	if !r.config.RetainTempDir {
+		defer os.RemoveAll(tempDir)
+	}
+
+	// Create build.gradle and settings.gradle
+	temporalSDKDependency := ""
+	if r.config.Version != "" {
+		temporalSDKDependency = fmt.Sprintf("implementation 'io.temporal:temporal-sdk:%v'", r.config.Version)
+	}
+	buildGradle := `
+plugins {
+    id 'application'
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'io.temporal:sdk-features:0.1.0'
+    ` + temporalSDKDependency + `
+}
+
+application {
+    mainClass = 'io.temporal.sdkfeatures.Main'
+}`
```

Comment:
Nit: Consider using some external templates for these things (in all langs). Not a big deal so feel free to ignore but we should probably do if we end up with much more stuff like this.

Reply:
If it gets more complicated, definitely. In this case and in Go's case it wasn't too bad inline. If we get any more complicated, for sure, though the goal is to do the smallest amount possible to "inherit from main project, but only change the SDK version" so hopefully it doesn't grow past this.